sklearn/utils/estimator_checks.py

```diff
@@ -331,6 +332,44 @@ def _construct_instance(Estimator):
     return estimator
 
 
+def _get_xfail_checks(estimator):
+    """Get xfail_check from estimator"""
+    if isinstance(estimator, type):
+        # try to construct estimator instance, if it is unable to then
+        # return then ignore xfail tag
+        try:
+            estimator = _construct_instance(estimator)
+        except Exception:
+            return {}
+
+    if not hasattr(estimator, "_get_tags"):
+        return {}
+
+    return estimator._get_tags()['_xfail_checks'] or {}
+
+
+def _check_warns_on_fail(estimator, check, xfail_checks_tag):
+    """Convert assertion errors to warnings if reason is given"""
```

Comment:
We can probably provide more details here:



```suggestion
	"""
    Wrap the check so that a warning is raised:
		- when the check is in the `xfail_checks` tag and the check properly failed as expected
		- when the check is in the `xfail_checks` tag and the check didnt fail but should have
		
	Checks that aren't in the xfail_checks tag aren't wrapped and are returned as-is.
	
	This wrapper basically simulates what pytest would do with the @xfail decorator, but this one can be used with check_estimator() which doesn't rely on pytest.
	"""
```